storage/merkle/tree.go

```diff
@@ -273,6 +277,100 @@ GetLoop:
 	}
 }
 
+// Prove constructs an inclusion proof for the given key, provided the key exists in the trie.
+// It returns a proof and a boolean (true if key exist and false if key not found)
+// Proof is constructed by traversing the trie from top to down and collects data for proof as follows:
+//  - if full node, append the sibling node hash value to sibling hash list
+//  - if short node, appends the node.shortCount to the short count list
+//  - if leaf, would capture the leaf value
+func (t *Tree) Prove(key []byte) (*Proof, bool) {
+
+	// check the len of key first
+	if t.keyLength != len(key) {
+		return nil, false
+	}
+
+	// we start at the root again
+	cur := &t.root
+
+	// and we start at a zero index in the path
+	index := 0
+
+	// init proof params
+	siblingHashes := make([][]byte, 0)
+	shortPathLengths := make([]uint16, 0)
+
+	steps := 0
+	shortNodeVisited := make([]bool, 0, len(key))
```

Comment:
Not sure what is the reason of allocating `len(key)` for the short node visited slice but not for the others? 
If there is no specific reason, it would make sense to make the approximative allocation for all the 3 slices or for nothing, just for consistency. 

`len(key)` is also not the maximum length the slice could have, the maximum would be the bit size of the key I think, `len(key)>>3`.

Reply:
I think this is a leftover from the past, we could definitely update it.